### PR TITLE
qcengine: init at 0.19.0

### PR DESCRIPTION
**Pull request by sheepforce**
###### Motivation for this change
Adds QCEngine python library. This is a scientific library, providing abstract python wrapper tools to computational chemistry packages. Part of the packages @markuskowa and me want to merge from markuskowa/NixOS-QChem#56.

###### Things done
- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Added a release notes entry if the change is major or breaking
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/python-modules/qcengine/default.nix [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+
+  src = fetchPypi {
+    inherit pname version;
+    sha256 = "0lz9r0fh31mcixdhayiwfc69cp8if9b3nkrk7gxdrb6vhbfrxhij";
+  };
+
+  doCheck = true;
+
+  meta = with lib; {
+    description = "Quantum chemistry program executor and IO standardizer (QCSchema) for quantum chemistry.";
```
Comment (markuskowa) [inline pkgs/development/python-modules/qcengine/default.nix:31]:
```suggestion
    description = "Quantum chemistry program executor and IO standardizer (QCSchema) for quantum chemistry";
```
(see here: https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md)

**Comment by markuskowa** [comment]
@GrahamcOfBorg build python3Packages.qcengine

**Comment by r-rmcgibbo** [comment]
Result of `nixpkgs-review pr 124501` at 3ea39147 run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages built successfully:</summary>
  <ul>
    <li>python38Packages.qcengine</li>
    <li>python39Packages.qcengine</li>
  </ul>
</details>

File: pkgs/development/python-modules/qcengine/default.nix [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+    sha256 = "0lz9r0fh31mcixdhayiwfc69cp8if9b3nkrk7gxdrb6vhbfrxhij";
+  };
+
+  doCheck = true;
+
+  meta = with lib; {
+    description = "Quantum chemistry program executor and IO standardizer (QCSchema) for quantum chemistry";
+    homepage = "http://docs.qcarchive.molssi.org/projects/qcelemental/en/latest/";
+    license = licenses.bsd3;
+    platforms = platforms.linux;
```
Comment (markuskowa) [inline pkgs/development/python-modules/qcengine/default.nix:34]:
```suggestion
    platforms = [ "x86_64-linux" ];
```
Ofborg fails to build it on "aarch64-linux"

Comment (sheepforce) [reply pkgs/development/python-modules/qcengine/default.nix]: I guess this makes sense. After all, half of the software packages it wraps are only available on x86-64. Has been changed.

**Pull request merged**